src/Emogrifier.php

```diff
@@ -268,6 +268,15 @@ class Emogrifier
      */
     private $debug = false;
 
+    /**
+     * Holds the eventually injected CSS concatenator dependency or the "JIT-instantiated" default implementation.
+     * Do not access this property directly, use $this->resolveCssConcatenator() instead, where you need access to the
+     * CSS concatenator object.
+     *
+     * @var Emogrifier\CssConcatenator
+     */
+    private $cssConcatenator = null;
```

Comment:
As I understand, we have subsequently agreed on a simpler solution, which is just to add a `createCssConcatenator` method that can be overridden to return a different or extended implementation.